classes/user/Repository.inc.php

```diff
@@ -230,20 +230,20 @@ public function getAccessibleStageRoles($userId, $contextId, &$submission, $stag
         // If unassigned, only managers and admins have access
         if (empty($accessibleStageRoles)) {
             $roleDao = DAORegistry::getDAO('RoleDAO'); /** @var RoleDAO $roleDao */
-            $userRoles = $roleDao->getByUserId($userId, $contextId);
-            foreach ($userRoles as $userRole) {
-                if (in_array($userRole->getId(), [Role::ROLE_ID_SITE_ADMIN, Role::ROLE_ID_MANAGER])) {
-                    $accessibleStageRoles[] = $userRole->getId();
-                }
+            if ($roleDao->userHasRole($contextId, $userId, Role::ROLE_ID_MANAGER)) {
+                $accessibleStageRoles[] = Role::ROLE_ID_MANAGER;
+            }
+            if ($roleDao->userHasRole(CONTEXT_SITE, $userId, Role::ROLE_ID_SITE_ADMIN)) {
+                $accessibleStageRoles[] = Role::ROLE_ID_SITE_ADMIN;
```

Comment:
`->userHasRole()` makes a separate DB request each time, so it's not the end of the world but it will make it less performant.

Reply:
Yep, the tricky thing here is that we're using `$contextId` for one and `CONTEXT_SITE` for the other. It'll be two hits to a unique index, though, so it should be OK.